src/sentry/api/endpoints/project_details.py

```diff
@@ -646,7 +646,8 @@ def put(self, request: Request, project) -> Response:
         if "dynamicSampling" in result:
             raw_dynamic_sampling = result["dynamicSampling"]
             fixed_rules = self._fix_rule_ids(project, raw_dynamic_sampling)
-            project.update_option("sentry:dynamic_sampling", fixed_rules)
+            if project.update_option("sentry:dynamic_sampling", fixed_rules):
+                changed_proj_settings["sentry:dynamic_sampling"] = result["dynamicSampling"]
             changed_proj_settings["sentry:dynamic_sampling"] = result["dynamicSampling"]
```

Comment:
Remove this line please as it is repeated

Reply:
just did 😉